### ISSUE TITLE
A request

### ISSUE THREAD

**Issue by Funky445** [Bungie-net/api#1787]
Esteemed Bungie API Team,

First and foremost, I want to congratulate and praise the job you all do. The fact that Bungie shares an API with their players and allows the creation of many third-party tools that directly interface with the game is something that is rarely seen in the industry. I applaud you for that.

There is just one slight “issue”.

Many of these tools that were created are being used in malicious or unintentional ways. Those ways unfortunately are promoting toxicity in the community. The biggest offender is by far raid.report. As you probably know, raid.report is a tool used for tracking players raid stats, most notably completions, and achievements like low mans and flawless completions. While the tool works wonderfully when used correctly, it also has nasty side effects. One of those side effects is creating an elitist and egotistical personality in some people. They ego their stats with little regard for other people. Some take it a step further and require certain achievements to be deemed worthy of playing with them. This happened for example after the contest extension for vow was announced. Many players, especially those that completed it in the first 24 hours, were displaying toxic behavior towards their peers who did it between hours 24 and 48. Now, with the upcoming Lightfall raid contest mode being 48 hours, I fear this behavior will repeat. I already have heard reports of people who are going to help players get the raid between 24-48 hours for the specific purposing of going them afterwards. To fix this problem, I have a couple suggestions:

1.	Hide the amount of people that completed a specific raid run on the API. For example, every raid completion always shows 1 person.
2.	Make it impossible to distinguish contest mode clears from the first day to the second day. This could be done by revoking the “period” (date) data from the API.
3.	Outright hide/make inaccessible all raid stats from the API to third party tools. This would allow stats to be shown in game, but prevent sites like raid.report from using it.

I would like to applaud the excellent work once again you do, just please be mindful of how the tools could be used in the wrong way.

Best,
A concerned player

**Comment by Aryathel** [comment]
Hello!

I wanted to chime in and give my two cents on this matter, as it is more of an opinion piece rather than an actual API issue.

With regards to raid report itself, it is by no means being used for "malicious" intent. It is not harming anyone or providing any problematic services. All it is is displaying publicly available information in a formatted way. How it is then used and interpreted after that point is no fault of the raid.report site, or the Bungie API.

I would argue that this is an issue of the individuals displaying this gatekeeping behavior, which I feel it is necessary to conclude that it is out of our control. Even if this information was removed from the Bungie API, there are still many ways to access the same information in game, which means that the same behavior would continue in a different way.

If you start to knock all those ways of comparison off, then suddenly we lose the ability to see a lot of different cool stats whatsoever, which leads to, in my opinion, a very non-competitive environment, which leads to a significant loss in engagement with the content.

Tl;dr - I think that this issue is the fault of the individuals involved, and is not an inherently toxic behavior to begin with, and that making the proposed changes would not solve the original problem statement.

**Comment by lvindotexe** [comment]
Can't hold the devs responsible for how people want to use their tools

**Comment by justrealmilk** [comment]
![dancing-duckdancing 1](https://user-images.githubusercontent.com/156681/219948614-761aecff-aa92-4216-995a-aa3ae8503483.gif)

**Comment by Funky445** [comment]
Hello Aryathel,

Firstly, thank you for your reply. I appreciate a thoughtful discussion.

As a counterargument to your point, I want to go back to 2000s gaming. During that time, most games like Halo used opt-out voice comms in multiplayer matches. You were able to freely talk to the opposing team when you matchmake with them. However, certain individuals started harassing others using these opt-out comms. As such, with the 2010s came a change nearly all multiplayer games adopted, they made the opt-out voice comms opt-in. Destiny included. This change was done precisely to combat harassment online, though it did not target the root cause of the problem: toxic individuals. I see the same thing happening with raid report. The information there is not at all malicious, but it can be used as such. Just like opt-out voice comms; they were not malicious in nature but could be used with malicious intent.

There would still be many ways to access information currently on raid report in game, but this in my opinion would not be nearly as toxic or harmful as it is today. Firstly, not being able to tell the difference between a "day 1" and a "contest" clear would already solve the issue that made me create this thread anyway, since there would be no way to do so in game. Toxic behavior would continue to exist, but it would be diminished in some capacity, which is always a good thing.

I agree that competition is good, but it cannot come at the cost of promoting a toxic environment. Furthermore, I would argue that while it would indeed decrease engagement for some, this would be more than offset by the increase in engagement that others get by having their stats hidden. For example, some people may get bullied by having a day 2 contest clear instead of a day 1 clear. However, those people would be benefitted of having this stat hidden because those that bully them for it would not be able to do it, which would increase engagement.

Lastly, out of the 3 suggestions I presented, I definitely think suggestion 2 would be the best. It would attack the issue that compelled me to write this out while leaving other parts of the information on raid report intact. Maybe instead of outright hiding period, the stat could be in some way modified to show the person beat the raid in the first 48 hours, but not when, which would once again solve the issue that brought me here.

**Comment by justrealmilk** [comment]
It doesn't matter what you do, the kind of player you wish to deter will always find a way to disrupt, exclude, or spread vile. This is true of all things and is in no way limited to video gaming–they always find a way. 

You build optimistically and never let a hateful minority stand in the way. There'll always be a crack to patch.

**Comment by nev-r** [comment]
it's certainly not wrong to want a better and kinder world, but your proposal does nothing to limit toxicity.
hide day 1 and someone could be just as weird about making sure you completed it by the end of day 2.
hide pgcr dates and the major scrapers will just determine date by when the pgcr appeared.
hide all raid launch information and someone could simply demand to see your special emblem.
hide a bunch of other information and it ruins why people want information about their activity history.

the contest mode/day 1 distinction is a remarkably specific thing to focus on.
people can be judgemental towards your K/D, your hours played, the weapons you have, the armor you have, the collection you have, your pvp skill ranking, soon they will your commendations and guardian level to gatekeep with.
it sounds like the day 1 issue has maybe affected you very personally?
sorry if people have been unkind but not everyone will scour your data to find something wrong.

if you want to opt out of people being able to query your activity history, you can do so in your bungie.net profile.
raid report actually **goes the extra mile** and **doesn't display your personal report page** if you are set to private,
**even though all pgcrs are public data and they have the information anyway.**

there's no "right answer" for when good things should be taken away, to prevent people from doing bad things, but you'll probably find the majority of people disagree with where you draw the line on this.
human attitude problems can have structural and institutional solutions, but this one is not fixed by blurring two days a year in the API.

and please remember that in the case of raiding, at least you ultimately have the choice of the players you interact with.
clans, friend groups, destiny is a good game to play with people you like.

**Comment by Funky445** [comment]
Hello Justrealmilk and nev-r,

Firstly addressing Justrealmilk, yes there will always be a way. But it doesn't mean anyone should give up. Players will find a way yes but any time they spend finding a way is less time they spend bullying people. Just like cheaters. They still want to cheat, but that doesn't mean devs shouldn't develop anti cheat.

As for nev-r, I do indeed know that bungie has a way to prevent websites from getting data. However, the current setting is very "all or nothing". From what I understand, either all third party websites use the data, or none at all. But you did give me a new idea.

What if there was a way to "blacklist" certain websites from displaying your data, while leaving other intact? For example, I could set my raid report to private but still allow other websites like bray tech to be able to function properly? I have no idea if this is even possible but I am willing to go the extra mile to think of as many solutions as necessary to attempt to mitigate this problem.

**Comment by jshaffstall-bng** [comment]
I understand that there can be 'elitism' problems that can arise from LFG and checking player bona fides, but we aren't planning to make any major changes to the related API systems for Destiny 2.
